resources/NBTest/NBTest_070_TestCarbonTradeSplitter.py

```diff
@@ -92,598 +63,467 @@ def get_tokens(self):
+                'tknout': WBTC_ADDRESS,
+                'amtin': 5,
+                'amtout': 0.5,
+                '_amtin_wei': 5000000000000000000,
+                '_amtout_wei': 50000000,
+            },
+            {
+                'cid': CID2,
+                'tknin': ETH_ADDRESS,
+                'tknout': WBTC_ADDRESS,
+                'amtin': 5,
+                'amtout': 0.5,
+                '_amtin_wei': 5000000000000000000,
+                '_amtout_wei': 50000000,
+            },
+        ]
+    ),
+)
 
+trade_instruction_12 = TradeInstruction(
+    cid=CID5,
+    tknin=WBTC_ADDRESS,
+    tknout=WETH_ADDRESS,
+    amtin=1,
+    amtout=10,
+    _amtin_wei=100000000,
+    _amtout_wei=10000000000000000000,
+    tknin_dec_override=8,
+    tknout_dec_override=18,
+    tknin_addr_override=WBTC_ADDRESS,
+    tknout_addr_override=WETH_ADDRESS,
+    exchange_override=PANCAKESWAP_V2_NAME,
+    ConfigObj=cfg,
+    db=db,
+    raw_txs=dumps([]),
+)
 
+trade_instruction_13 = TradeInstruction(
+    cid=CID3,
+    tknin=BNT_ADDRESS,
+    tknout=USDC_ADDRESS,
+    amtin=15,
+    amtout=150,
+    _amtin_wei=15000000000000000000,
+    _amtout_wei=150000000,
+    ConfigObj=cfg,
+    db=db,
+    raw_txs=dumps(
+        [
+            {
+                'cid': CID3,
+                'tknin': BNT_ADDRESS,
+                'tknout': USDC_ADDRESS,
+                'amtin': 10,
+                'amtout': 100,
+                '_amtin_wei': 10000000000000000000,
+                '_amtout_wei': 100000000,
+            },
+            {
+                'cid': CID4,
+                'tknin': BNT_ADDRESS,
+                'tknout': USDC_ADDRESS,
+                'amtin': 5,
+                'amtout': 50,
+                '_amtin_wei': 5000000000000000000,
+                '_amtout_wei': 50000000,
+            },
+        ]
+    ),
+)
 
+trade_instruction_14 = TradeInstruction(
+    cid=CID6,
+    tknin=USDC_ADDRESS,
+    tknout=BNT_ADDRESS,
+    amtin=150,
+    amtout=15,
+    _amtin_wei=150000000,
+    _amtout_wei=15000000000000000000,
+    tknin_dec_override=6,
+    tknout_dec_override=18,
+    tknin_addr_override=USDC_ADDRESS,
+    tknout_addr_override=BNT_ADDRESS,
+    exchange_override=BANCOR_V2_NAME,
+    ConfigObj=cfg,
+    db=db,
+    raw_txs=dumps([]),
+)
 
+test_cases = [
+    {
+        'input': [trade_instruction_0, trade_instruction_3],
+        'expected_output': [trade_instruction_7, trade_instruction_8, trade_instruction_9]
+    },
+    {
+        'input': [trade_instruction_1, trade_instruction_3],
+        'expected_output': [trade_instruction_10, trade_instruction_9]
+    },
+    {
+        'input': [trade_instruction_2, trade_instruction_4],
+        'expected_output': [trade_instruction_11, trade_instruction_12]
+    },
+    {
+        'input': [trade_instruction_5, trade_instruction_6],
+        'expected_output': [trade_instruction_13, trade_instruction_14]
+    },
+]
+
+def test_split_carbon_trades():
```

Comment:
IMO it's good practice to avoid complex code, like loops, in test functions. Also, by writing the tests this way, when it runs using pytest, the entire loop of assertions gets grouped together, so you can't immediately understand which test case failed, or if more than one assertion would have failed. A simple way to avoid loops, as well as solve the grouped failure problem, is to use pytest fixtures. For example, in this particular case the pytest solution is as follows:  

```python 
import pytest

@pytest.fixture
def cfg_fixture():
    return Config()

# Assuming TradeInstruction and split_carbon_trades are defined somewhere above
@pytest.fixture(params=[
    {
        'input': [trade_instruction_0, trade_instruction_3],
        'expected_output': [trade_instruction_7, trade_instruction_8, trade_instruction_9],
    },
    {
        'input': [trade_instruction_1, trade_instruction_3],
        'expected_output': [trade_instruction_10, trade_instruction_9],
    },
    {
        'input': [trade_instruction_2, trade_instruction_4],
        'expected_output': [trade_instruction_11, trade_instruction_12],
    },
    {
        'input': [trade_instruction_5, trade_instruction_6],
        'expected_output': [trade_instruction_13, trade_instruction_14],
    },
])
def test_case(request):
    return request.param

def test_split_carbon_trades(test_case, cfg_fixture):
    input_trades = test_case['input']
    expected_output = test_case['expected_output']
    actual_output = split_carbon_trades(cfg_fixture, input_trades)
    assert actual_output == expected_output
```

Reply:
Fixed